packages/terra-compact-interactive-list/src/utils/keyHandlerUtils.js

```diff
@@ -0,0 +1,279 @@
+const focusableElementSelector = 'a[href]:not([tabindex=\'-1\']), area[href]:not([tabindex=\'-1\']), input:not([disabled]):not([tabindex=\'-1\']), '
+        + "select:not([disabled]):not([tabindex='-1']), textarea:not([disabled]):not([tabindex='-1']), button:not([disabled]):not([tabindex='-1']), "
+        + "iframe:not([tabindex='-1']), [tabindex]:not([tabindex='-1']), [contentEditable=true]:not([tabindex='-1'])";
+
+export const getFocusableElements = (parentElement) => [...parentElement.querySelectorAll(`${focusableElementSelector}`)].filter(
+  element => !element.hasAttribute('disabled')
+      && !element.getAttribute('aria-hidden')
+      && !!(element.offsetWidth || element.offsetHeight || element.getClientRects().length)
+      && window.getComputedStyle(element).visibility !== 'hidden'
+      && element.closest('[inert]') === null,
+);
+
+/**
+ * @param {HTMLElement} element - The element to check if it is a text input
+ * @returns True if the element is a text input. Otherwise, false.
+ */
+export const isTextInput = (element) => {
+  const { tagName } = element;
+  if (tagName.toLowerCase() === 'input') {
+    const validTypes = ['text', 'password', 'number', 'email', 'tel', 'url', 'search', 'date', 'datetime', 'datetime-local', 'time', 'month', 'week'];
+    const inputType = element.type;
+    return validTypes.indexOf(inputType) >= 0;
+  }
+
+  return false;
+};
+
+/**
+ * Move focus to next focusable element outside the container element
+ */
+export const moveFocusFromElement = (containerRef, id, moveForward) => {
```

Comment:
This is being duplicated from the data-grid package.  Do we need to put this logic in a common file somehow?

Reply:
That might be a good idea. I was thinking of it, but wasn't sure where exactly to put it so that data grid and compact interactive list both can consume it.